authors/apps/article_rating/views.py

```diff
@@ -0,0 +1,62 @@
+from django.db.models import Avg
+from django.core.exceptions import ObjectDoesNotExist
+from rest_framework import status
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from authors.apps.articles.models import Article
+from authors.apps.article_rating.models import ArticleRating
+from authors.apps.article_rating.serializers import ArticleRatingSerializer
+
+class AverageArticleRatingViewSet(ModelViewSet):
+    """Endpoint for getting the average rating of a given article"""
+    permission_classes = (IsAuthenticated,)
+    queryset = ArticleRating.objects.all()
+    serializer_class = ArticleRatingSerializer
+
+    def list(self, request, slug):
+        article = Article.objects.get(slug=slug)
+        queryset = ArticleRating.objects.all().filter(article=article)
+        serializer = ArticleRatingSerializer(queryset, many=True,context={'request': request})
+        average = queryset.aggregate(Avg('rating'))
+        return Response({"rating": average.get("rating__avg")})
+
+
+class ArticleRatingViewSet(ModelViewSet):
+    """Api endpoint for adding, editting and viewing article ratings"""
+    permission_classes = (IsAuthenticated,)
+    queryset =  ArticleRating.objects.all()
+    serializer_class = ArticleRatingSerializer
+
+    def create(self, request, *args, **kwargs):
+        article = Article.objects.get(slug=self.kwargs.get("slug"))
+        print(self.request.data)
```

Comment:
Hi, @llwasampijja lets remove this print statement

Reply:
I agree with @nadralia